### PR TITLE
impl Clone for BitIter

### PR DESCRIPTION
**Pull request by lcnr**
Required for WIP fix for https://github.com/slide-rs/specs/issues/593

### CODE REVIEW

File: src/iter/mod.rs [hunk 1]
```diff
@@ -20,6 +20,16 @@ pub struct BitIter<T> {
     pub(crate) prefix: [u32; LAYERS - 1],
 }
 
+impl<T: Clone> Clone for BitIter<T> {
+    fn clone(&self) -> Self {
+        BitIter {
+            set: self.set.clone(),
+            masks: self.masks.clone(),
+            prefix: self.prefix.clone(),
```
Comment (lcnr) [inline src/iter/mod.rs]: It is not required to explicitly clone masks and prefix, as both implement copy. I may remove this if undesired.

**Comment by WaDelma** [comment]
This cannot be just a derive?

**Comment by lcnr** [comment]
@WaDelma BitIter.set does not have a clone bound.

**Comment by lcnr** [comment]
oh, it actually does work :scream: sorry, my bad

**Comment by WaDelma** [comment]
Np. Derive automatically adds `Clone` bounds for type variables.

**Comment by lcnr** [comment]
this should be ready for merge

**Comment by WaDelma** [comment]
bors r+

Ah yeah I forgot to actually merge

**Pull request merged**